arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java

```diff
@@ -111,33 +108,16 @@ public void writeToFile(final Path filename) {
     maybeThrowTracingExceptions();
 
     final List<Module> modules = hub.getModulesToTrace();
-    final List<ColumnHeader> traceMap =
-        modules.stream().flatMap(m -> m.columnsHeaders().stream()).toList();
-    final int headerSize = traceMap.stream().mapToInt(ColumnHeader::headerSize).sum() + 4;
+    final List<Trace.ColumnHeader> headers =
+        modules.stream().flatMap(m -> m.columnHeaders().stream()).toList();
 
     try (RandomAccessFile file = new RandomAccessFile(filename.toString(), "rw")) {
-      file.setLength(traceMap.stream().mapToLong(ColumnHeader::cumulatedSize).sum());
-      final MappedByteBuffer header =
-          file.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, headerSize);
-
-      header.putInt(traceMap.size());
-      for (ColumnHeader h : traceMap) {
-        final String name = h.name();
-        header.putShort((short) name.length());
-        header.put(name.getBytes());
-        header.put((byte) h.bytesPerElement());
-        header.putInt(h.length());
-      }
-      long offset = headerSize;
+      Trace trace = Trace.of(file, headers, new byte[0]);
```

Comment:
@letypequividelespoubelles Here `new byte[0]` is for empty metadata.  In a future PR, I will populate this will actual information from the `zkevm.bin` file.